### PR TITLE
Use new payments within TransactionBuilder

### PR DESCRIPTION
**Pull request by dcousens**
Dependent on https://github.com/bitcoinjs/bitcoinjs-lib/pull/1096, review and merge that first.

Then I'll rebase this and remove the `no merge` tag.

### CODE REVIEW

**Comment by dcousens** [comment]
So close!
1 test failing

File: test/fixtures/transaction_builder.json [hunk 1]
```diff
@@ -1313,8 +1313,8 @@
         "locktime": 0
       },
       {
-        "description": "P2WPKH",
-        "txHex": "0100000000010133defbe3e28860007ff3e21222774c220cb35d554fa3e3796d25bf8ee983e1080000000000ffffffff0160ea0000000000001976a914851a33a5ef0d4279bd5854949174e2c65b1d450088ac0248304502210097c3006f0b390982eb47f762b2853773c6cedf83668a22d710f4c13c4fd6b15502205e26ef16a81fc818a37f3a34fc6d0700e61100ea6c6773907c9c046042c440340121038de63cf582d058a399a176825c045672d5ff8ea25b64d28d4375dcdb14c02b2b00000000",
+        "description": "P2WPKH -> P2PKH",
+        "txHex": "0100000000010133defbe3e28860007ff3e21222774c220cb35d554fa3e3796d25bf8ee983e1080000000000ffffffff0160ea0000000000001976a914851a33a5ef0d4279bd5854949174e2c65b1d450088ac02483045022100834f56825e880ab7926164458e10582d9fd8df005396b7e51a1efb8db277204e02206a3610b7101c3242643ac9c9d3487c2d28ffdad19ec26a7f81fc100bdac625f10121038de63cf582d058a399a176825c045672d5ff8ea25b64d28d4375dcdb14c02b2b00000000",
```
Comment (dcousens) [inline test/fixtures/transaction_builder.json]: TBD: why are these different

File: test/fixtures/transaction_builder.json [hunk 2]
```diff
@@ -395,7 +395,7 @@
       },
       {
         "description": "Transaction w/ P2WPKH -> P2WPKH",
-        "txHex": "01000000000101ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff0000000000ffffffff011027000000000000160014aa4d7985c57e011a8b3dd8e0e5a73aaef41629c502483045022100a8fc5e4c6d7073474eff2af5d756966e75be0cdfbba299518526080ce8b584be02200f26d41082764df89e3c815b8eaf51034a3b68a25f1be51208f54222c1bb6c1601210279be667ef9dcbbac55a06295ce870b07029bfcdb2dce28d959f2815b16f8179800000000",
+        "txHex": "01000000000101ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff0000000000ffffffff011027000000000000160014aa4d7985c57e011a8b3dd8e0e5a73aaef41629c502483045022100b4a9d46ea4d38d6b3ea098911c9f72c0ae6ebc72408e6be7880a6b22a4b3e4da02207996107d0e6437f80363f96f502a38f275156f7501ea51f67899ba78a0c129c101210279be667ef9dcbbac55a06295ce870b07029bfcdb2dce28d959f2815b16f8179800000000",
```
Comment (dcousens) [inline test/fixtures/transaction_builder.json]: TBD: why are these different

Comment (dcousens) [reply test/fixtures/transaction_builder.json]: I think they may have been wrong... the integration tests pass?

File: test/fixtures/transaction_builder.json [hunk 3]
```diff
@@ -833,8 +833,8 @@
...
+        "description": "SIGHASH V0+V1, (P2PKH, P2WPKH) -> 2x P2PKH",
+        "txHex": "01000000000102fff7f7881a8099afa6940d42d1e7f6362bec38171ea3edf433541db4e4ad969f00000000484730440220691a19d365c8d75f921346c70271506bde136f13a4b566dd796902c262e2ec6d02202b00c4aa030eedf294552bdfc163936d2f4e91c59e7798c4471250cf07cb859501eeffffffef51e1b804cc89d182d279655c3aa89e815b1b309fe287d9b2b55d57b90ec68a0100000000ffffffff0230f45e13000000001976a9148280b37df378db99f66f85c95a783a76ac7a6d5988ac00e9a435000000001976a9143bde42dbee7e4dbe6a21b2d50ce2f0167faa815988ac000247304402200a7b08cccedf608e279410091acbd7e990e19a8edf401c3698763d2920de5871022060462ed172a02ecef73ebc19811d8fc72ed68f4419742df70241ad0a5a6a36410121025476c2e83188368da1ff3e292e7acafcdb3566bb0ad253f62fc70f07aeee635711000000",
```
Comment (dcousens) [inline test/fixtures/transaction_builder.json]: And this

File: src/transaction_builder.js [hunk 4]
```diff
@@ -206,265 +158,262 @@ function fixMultisigOrder (input, transaction, vin) {
...
+    }
 
-    prevOutType = scriptTypes.P2PKH
-    witness = false
-    signType = prevOutType
-    signScript = prevOutScript
-  }
+    let signScript = redeemScript
+    if (expanded.type === SCRIPT_TYPES.P2WPKH) {
+      signScript = payments.p2pkh({ pubkey: expanded.pubkeys[0] }).output
```
Comment (dcousens) [inline src/transaction_builder.js]: BIP143 support here

**Comment by dcousens** [comment]
@afk11 you touched this code quite a bit,  if you have the time,  could you review too?

**Comment by junderw** [comment]
We can wait for @afk11 review, or you can merge, @dcousens

**Pull request merged**